waspc/data/Generator/templates/server/src/auth/providers/index.ts

```diff
@@ -0,0 +1,28 @@
+import { Router } from "express";
+
+import { getDirFromFileUrl, importJsFilesFromDir } from "../../utils.js";
+
+import { ProviderConfig } from "./types";
+
+const providers = await importProviders();
+
+const router = Router();
+
+for (const provider of providers) {
+  const { init, setupRouter } = provider;
+  let initData;
+  if (init) {
+    initData = await init(provider);
+  }
+  const providerRouter = setupRouter(provider, initData);
+  router.use(`/${provider.slug}`, providerRouter);
+  console.log(`🚀 "${provider.name}" auth initialized`)
+}
+
+export default router;
+
+async function importProviders(): Promise<ProviderConfig[]> {
```

Comment:
Dynamically import all the provider configs from the `./config` folder.

This enables to keep most of the code without templating, and configure enabled providers by either putting a config file in the `config` folder or not.